### PR TITLE
Fix deprecationwarnings issued by old uses of the representation= kwarg in coordinates

### LINKED ISSUE #8139: Find root cause of representation deprecation warnings all over coordinates
In the 3.1rc1, we are now getting deprecation warnings almost everywhere in coordinates.  I see it at `SkyCoord` creation, frame class creation, and `repr`, which are pretty fundamental.

This leads me to suspect we missed converting `representation` to `representation_type` somewhere deep in the machinery (like a getattr or something).  Just need to hunt it down.

Really, this is my bad from #8119... Worse case scenario we can just revert that PR before the final 3.1 release and delay the deprecation of the keyword for another version.

cc @adrn

### PR DESCRIPTION
**Pull request by adrn**
This fixes #8139.

### CODE REVIEW

**Comment by adrn** [comment]
The test failure is real, but I actually don't understand this test: https://github.com/astropy/astropy/blob/master/astropy/coordinates/tests/test_regression.py#L429

If you pass a `representation` instance to a frame, and specify the `representation_type`, shouldn't it change the representation type of the frame? So shouldn't we expect that test to fail? 

cc @mhvk

**Comment by eteq** [comment]
@adrn re: the test... I think the point there is that it already *is* a `UnitSphericalRepresentation`, so passing `representation_type='unitspherical'` should be a no-op.  Perhaps the problem is that it's now doing a `represent_as` which is unintentionally copying?

**Comment by mhvk** [comment]
@eteq is right - the point of the test was to ensure that if the representation was correct, no copy would done. I think of this like providing `dtype` to `asarray` - if already the right type, no need to copy.

**Comment by eteq** [comment]
I just added commits that do a tricky thing with pytest to inject a fixture that turns these deprecations into errors for *just* the coordinates tests (and fixes a bunch of left-over things that were still not using the new syntax). This will presumably be a moot point in the next version or two when we'd make this an error anyway, but it's a good way to get everything up-to-speed sooner rather than later.

(I also did a quick experiment of running this on the *whole* test suite, and included the fixes from that here too. I'm not sure we want to subject all subpackages to this restriction though, so I left the warning convesion in just coordinates.)

This doesn't fix https://github.com/astropy/astropy/pull/8140#issuecomment-439543190 though...

File: astropy/coordinates/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import pytest
+import warnings
+
+
+# autouse makes this an all-coordinates-tests fixture
+@pytest.fixture(autouse=True)
```
Comment (pllim) [inline astropy/coordinates/tests/conftest.py]: I guess this should be undone when #7928 is addressed (in the long run) and we can then turn most (if not all) warnings to errors?

Comment (eteq) [reply astropy/coordinates/tests/conftest.py]: Yes, good point.  will add a reference to #7928 in here so we have at least a chance of remembering.

**Comment by bsipocz** [comment]
the failure is coordinates related, though looks like to be a different, frame comparision kind. Could you have a look at it so we can merge this?

**Comment by adrn** [comment]
Strange - I can't think of what in this PR could be causing the failures. Any ideas @eteq ?

**Comment by eteq** [comment]
Isn't it all the same problem as https://github.com/astropy/astropy/pull/8140#issuecomment-439543190?

What I don't understand is why it ran even though I had ``[ci skip]`` in the comment of the last commit...?

**Comment by bsipocz** [comment]
@eteq - circleCI doesn't really respect `ci skip`, but appveyor will only respect it if it's in the main part of the commit message, and that isn't the case for your last one: https://www.appveyor.com/docs/how-to/filtering-commits/#skip-directive-in-commit-message

**Comment by bsipocz** [comment]
This is now passing and approved, let's see it fixes downstream, too.

**Comment by bsipocz** [comment]
Thanks @adrn!

**Pull request merged**